packages/dev/core/src/Debug/debugLayer.ts

```diff
@@ -261,18 +261,34 @@ export class DebugLayer {
         return this._onSelectionChangedObservable;
     }
 
+    /**
+     * the default configuration of the inspector
+     */
+    public config: IInspectorOptions = {
+        overlay: false,
+        showExplorer: true,
+        showInspector: true,
+        embedMode: false,
+        handleResize: true,
+        enablePopup: true,
+    };
+
     /**
      * Instantiates a new debug layer.
      * The debug layer (aka Inspector) is the go to tool in order to better understand
      * what is happening in your scene
      * @see https://doc.babylonjs.com/toolsAndResources/inspector
      * @param scene Defines the scene to inspect
+     * @param config the default configuration of the inspector
      */
-    constructor(scene?: Scene) {
+    constructor(scene?: Scene, config?: IInspectorOptions) {
         this._scene = scene || <Scene>EngineStore.LastCreatedScene;
         if (!this._scene) {
             return;
         }
+        if (config) {
+            this.config = config;

```

Comment:
This would force to always specify all the value. I guess placing the config static should also help preventing this issue.

Reply:
I have thought about this design before, for this kind of problem, my ideal design is generally static configuration items, there are configuration items on the instance, and the configuration items on the instance will inherit the static configuration items.

Reply:
For the multi-layer dynamic inheritance of instance configuration items, I also specially encapsulated the tool of configuration items, but I didn't plan to use that tool at this time, and I just made a one-time shallow copy